packages/hardhat-helpers/src/deploy.js

```diff
@@ -87,17 +87,29 @@ export const deployUpgradeableContract = async (
 
 export const verify = async ({ address, deployArgs, contract }) => {
   const { run } = require('hardhat')
-  try {
-    await run('verify:verify', {
-      address,
-      contract,
-      constructorArguments: deployArgs,
-    })
-  } catch (error) {
-    console.log(
-      `FAIL: Verification failed for contract at ${address} with args : ${deployArgs.toString()}`
-    )
-    console.log(error)
+  let tries = 0
```

Comment:
I went for 5 tries approach so it's not too verbose and still exits after it failed too many times.